src/CommandProcessor.php

```diff
@@ -137,8 +141,19 @@ public function validateRunAndAlter(
             return $validated;
         }
 
-        // Run the command, alter the results, and then handle output and status
-        $result = $this->runCommandCallback($commandCallback, $commandData);
+        $replaceDispatcher = new ReplaceCommandHookDispatcher($this->hookManager(), $names);
+        if ($this->logger) {
```

Comment:
Er, well that's not true. I guess we can just have it match? I liked using a single array, but we can write less code the other way. Let me know which you'd prefer.